### ISSUE TITLE
Named Injections not working with current VajramGuiceInjector

### ISSUE THREAD

**Issue by shivangipandey** [flipkart-incubator/Krystal#285]
existing Method : 

 ```
 private @Nullable Annotation getQualifier(VajramID vajramID, InputDef<?> inputDef) {
    Optional<VajramDefinition> vajramDefinition = vajramKryonGraph.getVajramDefinition(vajramID);
    if (vajramDefinition.isEmpty()) {
      return null;
    }
    ImmutableMap<String, ImmutableMap<Object, Tag>> facetTags =
        **vajramDefinition.get().getFacetTags();** **// Coming as null**

    String inputName = inputDef.name();
    return Optional.ofNullable(
            facetTags.getOrDefault(inputName, ImmutableMap.of()).get(Named.class))
        .map(
            tag -> {
              if (tag instanceof AnnotationTag<?> annoTag
                  && annoTag.tagValue() instanceof Named named) {
                return named;
              }
              return null;
            })
        .orElse(null);
  }
```

**Comment by RamAnvesh** [comment]
`vajramDefinition.get().getFacetTags()` returning null
@shivangipandey Couple of questions:
1. Is `getFacetTags()` it returning null, or empty map?
2. Which vajram are you trying this for?

**Comment by shivangipandey** [comment]
vajramDefinition.get().getFacetTags() is returning null. 

I'm doing this for IO Vajram.

**Comment by RamAnvesh** [comment]
Understood. I'm working on it

**Comment by RamAnvesh** [comment]
Update: Tested this locally. There is a bug in 

https://github.com/flipkart-incubator/Krystal/blob/00096805bf9d811767a6ff3c220e6799f4f2efd3/vajram/src/main/java/com/flipkart/krystal/vajram/exec/VajramDefinition.java#L69

The `parseFacetTags` method is reading facet fields from the vajramImpl (wrapper) class instead of the `_Facets` class. This needs to be fixed.

But in my tests `vajramDefinition.get().getFacetTags()` is returning an empty map instead of null. Will Fix this first and see if the original issue gets fixed

**Comment by RamAnvesh** [comment]
@shivangipandey This issue should be fixed in the latest 8.0.0-SNAPSHOT. Please try it out.
An existing unit test which had named injection had a bug because of which this was not caught. Fixed that test as well
